src/Components/ServiceScene/LogsTableScene.tsx

```diff
@@ -154,29 +233,44 @@ export class LogsTableScene extends SceneObjectBase<LogsTableSceneState> {
           showMenuAlways={true}
           actions={
             <>
-              <Button onClick={() => model.showColumnManagementDrawer(true)} variant={'secondary'} size={'sm'}>
-                Manage columns
-              </Button>
+              {!logsControlsSupported && (
+                <Button onClick={model.onManageColumnsClick} variant={'secondary'} size={'sm'}>
+                  Manage columns
+                </Button>
+              )}
               <LogsPanelHeaderActions vizType={visualizationType} onChange={parentModel.setVisualizationType} />
             </>
           }
         >
-          {dataFrame && (
-            <TableProvider
-              panelWrap={panelWrap}
-              addFilter={addFilter}
-              timeRange={timeRangeValue}
-              selectedLine={selectedLine}
-              urlColumns={urlColumns ?? []}
-              setUrlColumns={setUrlColumns}
-              dataFrame={dataFrame}
-              clearSelectedLine={clearSelectedLine}
-              setUrlTableBodyState={setUrlTableBodyState}
-              urlTableBodyState={tableLogLineState}
-              showColumnManagementDrawer={model.showColumnManagementDrawer}
-              isColumnManagementActive={isColumnManagementActive}
-            />
-          )}
+          <div className={styles.container}>
+            {logsControlsSupported && (
+              <LogListControls
+                sortOrder={sortOrder}
+                onSortOrderChange={model.handleSortChange}
+                onManageColumnsClick={model.onManageColumnsClick}
+                onLineStateClick={model.onLineStateClick}
+                // "Auto" defaults to display "show text"
```

Comment:
from what I remember auto will show text if the cardinality of the labels is 1, otherwise will show labels

Reply:
What I did was forcing both states (with labels and with no labels) in the component and check that both the Table and the Controls show the same icon.

https://github.com/grafana/logs-drilldown/blob/main/src/Components/Table/LogLineCellComponent.tsx#L136

Hopefully this was a correct interpreation of it.